tests/deploy/e2e/test_autostarted.py

```diff
@@ -315,3 +360,779 @@ async def test_autostart_clear_agent_venv_on_delete(
+    result = await client.release_version(environment, version, push=False)
+    assert result.code == 200
+
+    async def are_resources_deploying(deployed_resources: int = 1) -> bool:
+        result = await client.resource_list(environment, deploy_summary=True)
+        assert result.code == 200
+        summary = result.result["metadata"]["deploy_summary"]
+        deployed = summary["by_state"]["deploying"]
+        return deployed == deployed_resources
+
+    # Wait for this resource to be deployed
+    await retry_limited(are_resources_deploying, 5)
+
+    # Executors are reporting to be deploying before deploying the first executor, we need to wait for them to be sure that
+    # something is moving
+    await wait_for_consistent_children(
+        current_pid=current_pid,
+        should_scheduler_be_defined=True,
+        should_fork_server_be_defined=True,
+        executor_to_be_defined=1,
+    )
+
+    current_children_mapping = construct_scheduler_children(current_pid)
+    for children in current_children_mapping.children:
+        assert children.is_running()
+
+    # Retrieve the current processes, we should have more processes than `start_children`
+    children_after_deployment = construct_scheduler_children(current_pid)
+
+    result = await client.agent_action(tid=environment, name="agent1", action=AgentAction.pause.value)
+    assert result.code == 200
+
+    # Let's pretend that the server crashes
+    await asyncio.wait_for(server.stop(), timeout=20)
+    ibl = InmantaBootloader(configure_logging=False)
+    async_finalizer.add(partial(ibl.stop, timeout=20))
+    # Let's restart the server
+    await ibl.start()
+
+    # Wait for the scheduler
+    await wait_for_consistent_children(
+        current_pid=current_pid,
+        should_scheduler_be_defined=True,
+        should_fork_server_be_defined=False,
+        executor_to_be_defined=0,
+    )
+
+    # Everything should be consistent in DB: the agent should still be paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": True})
+
+    # Let's recheck the number of processes after restarting the server
+    state_after_restart = construct_scheduler_children(current_pid)
+    assert (
+        len(state_after_restart.children) == len(children_after_deployment.children) - 2
+    ), "Only this process should be present: The Scheduler!"
+    for children in state_after_restart.children:
+        assert children.is_running()
+
+    result = await client.resource_list(environment, deploy_summary=True)
+    assert result.code == 200
+    summary = result.result["metadata"]["deploy_summary"]
+    assert summary["total"] == 1, f"Unexpected summary: {summary}"
+    # FIXME this should be fixed -> old resource is still in deploying state, should be available
+    # Uncomment this once fixed, see https://github.com/inmanta/inmanta-core/issues/8216
+    # assert summary["by_state"]["available"] == 1, f"Unexpected summary: {summary}"
```

Comment:
Could you add a permalink to this part in the referenced ticket if it is still relevant ?

Reply:
It's already been fixed in another PR, so I think we don't need to create a ticket as it should be merged soon.